File: python/language.py
from transactions import Transactions

class Language:
    def __init__(self, language):
        self.language = language

    def answerTr(self):
        alternativesNames = []
        naturalStatesNames = []

        # number of alternative and natural situations
        numberOfAlternatives = int(input('Lütfen alternatif sayısını giriniz: '))
        numberOfNaturalStates = int(input('Lütfen doğal durum sayısını giriniz: '))

        print()
        # enter alternative name
        for i in range(0, numberOfAlternatives):
            result = input(f"{i + 1}. Alternatifinin ismini giriniz: ")
            alternativesNames.append(str(result))
            i += 1

        print()
        # enter natural situations
        for i in range(0, numberOfNaturalStates):
            result = str(input(f"{i + 1}. Doğal durumun ismini giriniz:: "))
            naturalStatesNames.append(result)

        print()
        tableCache = []
        table = []
        # enter table values
        for i in range(0, numberOfAlternatives):
            for j in range(0, numberOfNaturalStates):
                value = int(input(
                    f"[{alternativesNames[i]}] alternatifinin [{naturalStatesNames[j]}] doğal durumundaki değeri giriniz: "))
                tableCache.append(value)
            table.append(tableCache)
            tableCache = []
        
        print()
        # show table
        print("Tablo misali görünüm\n")
        counter = 0
        for item in table:
            print("\r" + alternativesNames[counter] + " --> " + str(item))
            counter += 1

        print()
        while True:
            choice = str(input("Probleminiz maliyet mi(E/H): "))
            if choice == "e" or choice == "E":
                transactionsCost = Transactions(language="tr", table=table, isCost=True,
                                            alternativesNames=alternativesNames)
                transactionsCost.optimism()
                transactionsCost.pessimism()
                transactionsCost.hurwicz()
                transactionsCost.laplace(numberOfNaturalStates=numberOfNaturalStates)
                break
            elif choice == "H" or choice == "h":
                transactionsProfit = Transactions(language=self.language, table=table, isCost=False,
                                            alternativesNames=alternativesNames)
                transactionsProfit.optimism()
                transactionsProfit.pessimism()
                transactionsProfit.hurwicz()
                transactionsProfit.laplace(numberOfNaturalStates=numberOfNaturalStates)
                break
            else:
                print("Lütfen geçerli değer giriniz!")

    def answerEn(self):
        alternativesNames = []
        naturalStatesNames = []

        # number of alternative and natural situations
        numberOfAlternatives = int(input('Please enter the number of alternatives: '))
        numberOfNaturalStates = int(input('Please enter the number of natural states: '))

        print()
        # enter alternative name
        for i in range(0, numberOfAlternatives):
            result = input(f"{i + 1}. Enter the name of the alternative: ")
            alternativesNames.append(str(result))
            i += 1

        print()
        # enter natural situations
        for i in range(0, numberOfNaturalStates):
            result = str(input(f"{i + 1}. Enter the name of the natural state: "))
            naturalStatesNames.append(result)

        print()
        tableCache = []
        table = []
        # enter table values
        for i in range(0, numberOfAlternatives):
            for j in range(0, numberOfNaturalStates):
                value = int(input(
                    f"[{alternativesNames[i]}] alternative [{naturalStatesNames[j]}] enter the value in its natural state: "))
                tableCache.append(value)
            table.append(tableCache)
            tableCache = []
        
        print()
        # show table
        print("Looks like your table\n")
        counter = 0
        for item in table:
            print("\r" + alternativesNames[counter] + " --> " + str(item))
            counter += 1
            
        print()
        while True:
            choice = str(input("Is the problem type cost?(Y/N): "))

            if choice == "y" or choice == "Y":
                transactionsCost = Transactions(language=self.language, table=table, isCost=True,
                                                alternativesNames=alternativesNames)
                transactionsCost.optimism()
                transactionsCost.pessimism()
                transactionsCost.hurwicz()
                transactionsCost.laplace(numberOfNaturalStates=numberOfNaturalStates)
                break
            elif choice == "N" or choice == "n":
                transactionsProfit = Transactions(language=self.language, table=table, isCost=False,
                                                  alternativesNames=alternativesNames)
                transactionsProfit.optimism()
                transactionsProfit.pessimism()
                transactionsProfit.hurwicz()
                transactionsProfit.laplace(numberOfNaturalStates=numberOfNaturalStates)
                break
            else:
                print("Please enter valid value!")

File: java/src/Language.java
import java.util.Scanner;

public class Language {

    public static void answersTr() {
        Scanner scanner = new Scanner(System.in);

        int numberOfAlternatives;
        int numberOfNaturalStates;

        // number of alternative and natural situations
        System.out.print("Lütfen alternatif sayısını giriniz: ");
        numberOfAlternatives = scanner.nextInt();
        System.out.print("Lütfen doğal durum sayısını giriniz: ");
        numberOfNaturalStates = scanner.nextInt();

        int[] alternatives = new int[numberOfAlternatives];
        int[] naturalStates = new int[numberOfNaturalStates];
        int[][] table = new int[numberOfAlternatives][numberOfNaturalStates];
        String[] alternativesNames = new String[numberOfAlternatives];
        String[] naturalStatesNames = new String[numberOfNaturalStates];

        // enter alternative name
        for (int i = 0; i < alternatives.length; i++)
        {
            System.out.print((i + 1) + " . Alternatifinin ismini giriniz: ");
            alternativesNames[i] = new Scanner(System.in).nextLine();
        }

        // entering natural situations
        for (int i = 0; i < naturalStates.length; i++)
        {
            System.out.print((i + 1) + " . Doğal durumun ismini giriniz: ");
            naturalStatesNames[i] = new Scanner(System.in).nextLine();
        }

        // enter table values
        for (int i = 0; i < numberOfAlternatives; i++)
        {
            for (int j = 0 ; j < numberOfNaturalStates; j++)
            {
                System.out.print("[" + alternativesNames[i] + "] alternatifi [" + naturalStatesNames[j] + "] doğal durumundaki değerini giriniz: ");
                table[i][j] = Integer.parseInt(new Scanner(System.in).nextLine());
            }
            System.out.println();
        }

        System.out.println("Tablonuz misali görünümü.");
        // show table
        for (int i = 0; i < numberOfAlternatives; i++) {
            System.out.print(alternativesNames[i]);
            for (int j = 0; j < numberOfNaturalStates; j++) {
                System.out.print(" " + table[i][j] + " ");
            }
            System.out.println();
        }

        String alternativeName = alternativesNames[0];

        while (true){

            System.out.print("Problem türü maliyet mi?(E/H): ");
            String answer = scanner.next();

            if (answer.equals("E") || answer.equals("e"))
            {
                Transactions.theCostOfOptimism(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                Transactions.theCostOfPessimism(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                Transactions.theCostOfEqualProbability(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                Transactions.theCostOfHurwicz(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                Transactions.savage(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                break;
            }
            else if (answer.equals("H") || answer.equals("h"))
            {
                Transactions.theProfitOfOptimism(numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                Transactions.theProfitOfPessimism(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                Transactions.theProfitOfEqualProbability(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                Transactions.theProfitOfHurwicz(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                Transactions.savage(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "TR");
                break;
            }else
                System.out.println("Lütfen geçerli değer giriniz!");
        }
    }
    public static void answersEn() {
        Scanner scanner = new Scanner(System.in);

        int numberOfAlternatives;
        int numberOfNaturalStates;

        // number of alternative and natural situations
        System.out.print("Please enter the number of alternatives:");
        numberOfAlternatives = scanner.nextInt();
        System.out.print("Please enter the number of natural states: ");
        numberOfNaturalStates = scanner.nextInt();

        int[] alternatives = new int[numberOfAlternatives];
        int[] naturalStates = new int[numberOfNaturalStates];
        int[][] table = new int[numberOfAlternatives][numberOfNaturalStates];
        String[] alternativesNames = new String[numberOfAlternatives];
        String[] naturalStatesNames = new String[numberOfNaturalStates];

        // enter alternative name
        for (int i = 0; i < alternatives.length; i++)
        {
            System.out.print((i + 1) + " . Enter the name of the alternative: ");
            alternativesNames[i] = new Scanner(System.in).nextLine();
        }

        // entering natural situations
        for (int i = 0; i < naturalStates.length; i++)
        {
            System.out.print((i + 1) + " . Enter the name of the natural state: ");
            naturalStatesNames[i] = new Scanner(System.in).nextLine();
        }

        // enter table values
        for (int i = 0; i < numberOfAlternatives; i++)
        {
            for (int j = 0 ; j < numberOfNaturalStates; j++)
            {
                System.out.print("[" + alternativesNames[i] + "] alternative [" + naturalStatesNames[j] + "] enter the value in its natural state: ");
                table[i][j] = Integer.parseInt(new Scanner(System.in).nextLine());
            }
            System.out.println();
        }

        System.out.println("Looks like your table.");
        // show table
        for (int i = 0; i < numberOfAlternatives; i++) {
            System.out.print(alternativesNames[i]);
            for (int j = 0; j < numberOfNaturalStates; j++) {
                System.out.print(" " + table[i][j] + " ");
            }
            System.out.println();
        }

        String alternativeName = alternativesNames[0];

        while (true){

            System.out.print("Is the problem type cost? (Y/N): ");
            String answer = scanner.next();

            if (answer.equals("Y"))
            {
                Transactions.theCostOfOptimism(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                Transactions.theCostOfPessimism(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                //Transactions.theCostOfEqualProbability(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                Transactions.theCostOfHurwicz(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                Transactions.savage(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                break;
            }
            else if (answer.equals("N"))
            {
                Transactions.theProfitOfOptimism(numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                Transactions.theProfitOfPessimism(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                Transactions.theProfitOfEqualProbability(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                Transactions.theProfitOfHurwicz(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                Transactions.savage(numberOfAlternatives, numberOfNaturalStates, table, alternativesNames, alternativeName, "EN");
                break;
            }else
                System.out.println("Please enter valid value!");
        }
    }
}